src/test/java/com/example/customerdb/integration/ApiIntegrationTest.java

```diff
@@ -0,0 +1,327 @@
+                      },
+                      {
+                         "price":12000
+                      }
+                ]                   
+                """, response, JSONCompareMode.STRICT);
+    }
+
+
+
+    @Test
+    @DataSet(value = "datasets/blankList.yml")
+    @ExpectedDataSet(value = "datasets/blankList.yml")
+    void 情報の取得APIを実行したときに情報がなければ空のリストを返却しステータスコードが200であること() throws Exception {
+        String response = mockMvc.perform(get("/purchase-info"))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+            JSONAssert.assertEquals("""  
+                [
+                
+                ]                   
+                """, response, JSONCompareMode.STRICT);
+        }
+
+    @Test
+    @DataSet(value = "datasets/infoList.yml")
+    @ExpectedDataSet(value = "datasets/insertInfoList.yml", ignoreCols = {"id", "purchaseDate"})
+    void 情報の登録APIを実行したときに正しく情報を登録しステータスコードが200であること() throws Exception {
+        PurchaseInfo newPurchaseInfo = new PurchaseInfo(0, "kato", "kto@example.com", null, 9000);
+        ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+        String requestBody = ow.writeValueAsString(newPurchaseInfo);
+
+            String response = mockMvc.perform(post("/purchase-info")
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(requestBody))
+                    .andExpect(status().isOk())
+                    .andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+            JSONAssert.assertEquals("""
+                {
+                    "message": "Info successfully created"
+                }
+                        """, response, JSONCompareMode.STRICT);
+    }
+
+    @Test
+    @Transactional
+    @DataSet(value = "datasets/infoList.yml")
+    @ExpectedDataSet(value = "datasets/updatedList.yml", ignoreCols = "id")
+    void PUTで指定したidの情報が更新できステータスコードが200であること() throws Exception {
+        PurchaseInfo updateInfo = new PurchaseInfo(1, "kinoshita", "knst@example.com", null, 2500);
+        ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+        String requestBody = ow.writeValueAsString(updateInfo);
+
+
+        String response = mockMvc.perform(put("/purchase-info/1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(requestBody))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+        JSONAssert.assertEquals("""
+                {
+                    "id":1,
+                    "purchaseInfo":{
+                       "id":1,
+                       "name":"kinoshita",
+                       "email":"knst@example.com",
+                       "purchaseDate":null,
+                       "price":2500
+                    }
+                }
```

Comment:
IDという情報が被っているのでこのようなレスポンスにしたいです。
```suggestion
                {
                    "purchaseInfo":{
                       "id":1,
                       "name":"kinoshita",
                       "email":"knst@example.com",
                       "purchaseDate":null,
                       "price":2500
                    }
                }
```
前コメントした気もしていますがレビューすりぬけちゃったかもですね。